engine/record_tx_repository.go

```diff
@@ -0,0 +1,53 @@
+package engine
```

Comment:
does it need to be in engine?

Reply:
I made a dao package.

But - in several places I have methods/functions/mocks/variables with the `repo` keyword. 
I'd like to adjust the naming in the next PR (after this one) to not adding additional changes here.